### PR TITLE
Improve the documentation on interoperability

### LINKED ISSUE #4869: Improve interoperability docs
https://docs.cupy.dev/en/latest/reference/interoperability.html

* Add PyTorch section
    * Move an example code from DLPack section
    * Add a link to memory pool sharing feature in pytorch-pfn-extras: https://github.com/pfnet/pytorch-pfn-extras/blob/master/docs/cuda.md.
    * Add an example of `cupy.asarray` and `torch.as_tensor` as PyTorch 1.3+ supports `__cuda_array_inferface__`
* Describe that `__cuda_array_inferface__` does not take care of ownership, unlike DLPack.
* Add TensorFlow as an example in DLPack section.

### PR DESCRIPTION
**Pull request by leofang**
Close #4869.

TODO:
- [x] Mention TF
- [x] Mention `pytorch-pfn-extras`?

### CODE REVIEW

**Comment by leofang** [comment]
@kmaehashi I think the only unaddressed action item is `pytorch-pfn-extras`. Would you like to push to my branch directly?

**Comment by leofang** [comment]
cc @harrism @jrhemstad Please review the mention of RMM if you have time, thanks!

**Comment by leofang** [comment]
FYI, the rendered page by the RTD CI is here: https://cupy--5023.org.readthedocs.build/en/5023/reference/interoperability.html

**Comment by kmaehashi** [comment]
Thank you! Pushed pytorch-pfn-extras docs.

**Comment by leofang** [comment]
Thanks, @kmaehashi! LGTM.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by leofang** [comment]
> Please review the mention of RMM if you have time, thanks!

Forgot to cc @jakirkham! 😄

**Comment by jakirkham** [comment]
cc @pentschev

File: docs/source/reference/interoperability.rst [hunk 1]
```diff
@@ -97,10 +106,123 @@ With the aforementioned ``__cuda_array_interface__`` standard implemented in CuP
...
+   >>> with ppe.cuda.stream(stream):
+   ...     ...
+
+
+RMM
+---
+
+`RMM (RAPIDS Memory Manager) <https://docs.rapids.ai/api/rmm/stable/index.html>`_ provides highly configurable memory allocators.
+
+RMM provides an interface to allow CuPy to allocate memory from its memory pool instead of from CuPy's own pool. It can be set up
```
Comment (harrism) [inline docs/source/reference/interoperability.rst:199]: (On mobile, can't use suggestion interface). Change "its memory pool" to "the RMM memory pool"

File: docs/source/reference/interoperability.rst [hunk 2]
```diff
@@ -97,10 +106,123 @@ With the aforementioned ``__cuda_array_interface__`` standard implemented in CuP
...
+as simple as follows:
```
Comment (harrism) [inline docs/source/reference/interoperability.rst:200]: ...as simply as...

**Review by harrism — COMMENTED** [review]
Minor suggestions for RMM section.

**Comment by leofang** [comment]
Thanks, @harrism!

**Comment by leofang** [comment]
Jenkins, test this please

**Review by kmaehashi — APPROVED** [review]
Great work! LGTM 👍🏼

**Pull request merged**